plugins/readtimer.koplugin/main.lua

```diff
@@ -66,26 +125,109 @@ function ReadTimer:remaining()
     end
 end
 
-function ReadTimer:remainingTime()
+-- can round
+function ReadTimer:remainingTime(round)
     if self:scheduled() then
         local remainder = self:remaining()
+        if round then
+            if round < 0 then -- round down
+                remainder = remainder - 59
+            elseif round == 0 then
+                remainder = remainder + 30
+            else -- round up
+                remainder = remainder + 59
+            end
+            remainder = math.floor(remainder * (1/60)) * 60
+        end
+
         local hours = math.floor(remainder * (1/3600))
         local minutes = math.floor(remainder % 3600 * (1/60))
         local seconds = math.floor(remainder % 60)
         return hours, minutes, seconds
     end
 end
 
+function ReadTimer:addAdditionalHeaderContent()
+    self.ui.crelistener:addAdditionalHeaderContent(self.additional_header_content_func)
+    self.update_status_bars(self, -1)
+end
+function ReadTimer:addAdditionalFooterContent()
```

Comment:
You can keep a blank line as saparator, we rarely stick them.
(May be I would have stick them, but not the "add" together, rather the Header together and the Footer together.)